Start FPGA getting started guide in target.md

AnotherButler opened the pull request:
Start getting started guide for the FPGA test shield.

Commit:
Start FPGA getting started guide in target.md

Start getting started guide for the FPGA test shield.

Commit:
Update FPGA section of target.md

Add more details about connecting the FPGA test shield.

AnotherButler commented:
ping @mprse

mprse commented on docs/porting/target/target.md:
````diff
@@ -85,3 +85,18 @@ The Mbed OS HAL provides a set of conformance tests for the Mbed OS HAL APIs. Yo
 ```
 mbed test -t <toolchain> -m <target> -n mbed-os-tests-mbed_hal*
 ```
+
````
if PR https://github.com/ARMmbed/mbed-os/pull/10820 will come in in current form, then:
`mbed-os-tests-mbed_hal*`  -> `tests-mbed_hal_fpga_ci_test_shield*`.

mprse replied:
I just realized that this line is for general HAL testing, so please ignore this comment.

mprse commented on docs/porting/target/target.md:
```diff
@@ -85,3 +85,18 @@ The Mbed OS HAL provides a set of conformance tests for the Mbed OS HAL APIs. Yo
+1. Add the `FPGA_CI_TEST_SHIELD` component to the Device Under Test in the `targets.json` configuration file.
+1. Run the test:
+
+   `mbed test -t GCC_ARM -m K66F -n tests-mbed_hal_fpga_ci_test_shield-spi_slave -v`
```
At the moment `spi_slave` will be available only on spi feature branch. I suggest selecting different test which will be available on master (e.g. `tests-mbed_hal_fpga_ci_test_shield-spi`).

mprse reviewed (approved):
Only 2 minor comments, otherwise ok.

mprse commented:
@jamesbeyond ping

Commit:
Change test in target.md

Update test code to code that points to master instead of a feature branch, as requested in comments.
